core/osm_connector.py

```diff
@@ -173,6 +173,8 @@ def get_routes(self, refresh=False):
                 self.routes[rv.ref] = rv
 
         # Cache data
+        if not  os.path.isdir('data'): #checking is the dir exist
+                os.mkdir('data')
         Cache.write_data('routes-' + self.selector, self.routes)
```

Comment:
I think this code would better be placed in the `Cache.write_data()` method, because this is where it logically belongs to.

style nitpick: two spaces after `not`.